### PR TITLE
feat: Checkpoint setter

### PR DESCRIPTION
**Pull request by akhercha**
Closes: #136 

### Added
- `checkpoint_setter` new service + integration tests

### Updated
- `pragma_sdk` few refinements (`__hash__` methods, naming etc...) + Checkpoint type
- `pragma_sdk->PragmaOnChainClient`: added new getters checkpoints methods + fixed set future checkpoints
- `price_pusher` Removed not needed methods now that we can hash dataclasses

### CODE REVIEW

File: vrf-listener/README.md [hunk 1]
```diff
@@ -34,9 +34,14 @@ Options:
...
+  -t, --check-requests-interval   Delay in seconds between checks for VRF
                                   requests. Defaults to 10 seconds.  [x>=0]
 
   --help                          Show this message and exit.
 ```
+
+Example:
+
+```sh
+
```
Comment (EvolveArt) [inline vrf-listener/README.md:46]: Empty

**Review by EvolveArt — COMMENTED** [review]
It's probably time we create the pragma-test package

File: checkpoint-setter/README.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                                  Delay in minutes between each new
+                                  checkpoints. Defaults to 60 minutes.  [x>=0]
+  
+  -help                          Show this message and exit.
+```
+
+For example:
+
+```sh
+python checkpoint_setter/main.py -c config/config.example.yaml --oracle-address $PRAGMA_ORACLE_ADDRESS --admin-address $PRAGMA_ADMIN_ACCOUNT -p plain:$MY_PRIVATE_KEY
```
Comment (EvolveArt) [inline checkpoint-setter/README.md:71]:
```suggestion
poetry run checkpointer -c config/config.example.yaml --oracle-address $PRAGMA_ORACLE_ADDRESS --admin-address $PRAGMA_ADMIN_ACCOUNT -p plain:$MY_PRIVATE_KEY
```

File: checkpoint-setter/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+classifiers = [
+  "Programming Language :: Python :: 3",
+  "License :: OSI Approved :: MIT License",
+  "Operating System :: OS Independent",
+]
+include = ["checkpoint_setter"]
+packages = [{ include = "checkpoint_setter" }]
+
+[tool.poetry.scripts]
+scheduler = 'checkpoint_setter.main:cli_entrypoint'
```
[the remark this answers is not in the document]
Comment (akhercha) [reply checkpoint-setter/pyproject.toml]: aye no strong opinion - updated to `checkpointer`

File: checkpoint-setter/README.md (hunk 2 above)
Comment (akhercha) [reply checkpoint-setter/README.md]: done

**Review by EvolveArt — APPROVED** [review]

**Pull request merged**